Sources/Web3Core/Utility/String+Extension.swift

```diff
@@ -135,6 +135,10 @@ extension String {
     func trim() -> String {
         trimmingCharacters(in: .whitespacesAndNewlines)
     }
+
+    public var isHex: Bool {
+        stripHexPrefix().reduce(true, { $0 && $1.isHexDigit } )
```

Comment:
Looks good when it's written in one line but we can improve it by returning false immediately when we encounter the first character that is not a hex digit.
This way we won't have to check all other characters.